webapp/src/js/services/contact-save.js

```diff
@@ -62,113 +62,103 @@ angular.module('inboxServices').service('ContactSave',
-            doc[fieldName] = original[fieldName];
+
+          if (preparedSibling.parent === 'PARENT') {
+            delete preparedSibling.parent;
+            // Cloning to avoid the circular reference we would make:
+            //   doc.fieldName.parent.fieldName.parent...
+            doc[fieldName] = _.clone(preparedSibling);
+            // Because we're assigning the actual doc referencem, the DB().get
+            // to attach the full parent to the doc will also attach it here.
+            preparedSibling.parent = doc;
           } else {
-            promiseChain = promiseChain.then(function() {
-              return DB().get(value).then(function(dbFieldValue) {
-                // In a correctly configured form one of these will be the
-                // parent This must happen before we attempt to run
-                // ExtractLineage on any siblings or repeats, otherwise they
-                // will extract an incomplete lineage
-                doc[fieldName] = extractIfRequired(fieldName, dbFieldValue);
-              });
-            });
+            doc[fieldName] = extractIfRequired(fieldName, preparedSibling);
           }
+
+          preparedSiblings.push(preparedSibling);
+        } else if (original &&
+                   original[fieldName] &&
+                   original[fieldName]._id === value) {
+          doc[fieldName] = original[fieldName];
+        } else {
+          promiseChain = promiseChain.then(function() {
+            return DB().get(value).then(function(dbFieldValue) {
+              // In a correctly configured form one of these will be the
+              // parent This must happen before we attempt to run
+              // ExtractLineage on any siblings or repeats, otherwise they
+              // will extract an incomplete lineage
+              doc[fieldName] = extractIfRequired(fieldName, dbFieldValue);
+            });
+          });
         }
       });
 
-      return promiseChain.then(function() {
-        return preparedSiblings;
-      });
-    }
+      return promiseChain.then(() => preparedSiblings);
+    };
 
-    return function(schema, form, docId, type) {
+    return function(form, docId, type) {
       return $q.resolve()
-        .then(function() {
-          if(docId) {
-            return DB().get(docId);
-          }
-          return null;
-        })
-        .then(function(original) {
-          var submitted = EnketoTranslation.contactRecordToJs(form.getDataStr({ irrelevant: false }));
-
-          if(original) {
+        .then(() => docId ? DB().get(docId) : null)
+        .then(original => {
+          const submitted = EnketoTranslation.contactRecordToJs(form.getDataStr({ irrelevant: false }));
+          if (original) {
             submitted.doc = $.extend({}, original, submitted.doc);
-          } else {
+          } else if (ContactTypes.isHardcodedType(type)) {
+            // default hierarchy - maintain backwards compatibility
```

Comment:
We're not defaulting to using `contact_type` and `type` for new contacts?
I was expecting all new contacts to use the new schema.

Reply:
As above, kept consistent for backwards compatibility with configuration.